### PR TITLE
Make region always mandatory in the KEB schema

### PR DESCRIPTION
**Pull request by MarekMichali**
**Description**

**Description**

Changes proposed in this pull request:

- Make schema creation more generic
- Generate schema with required `region` 
- Add a feature flag
- Adjust tests

**Related issue(s)**

See also https://github.com/kyma-project/control-plane/issues/3006

### CODE REVIEW

File: internal/appinfo/runtime_info.go [hunk 1]
```diff
@@ -30,20 +30,22 @@ type (
...
-		lastOperationFinder:     lastOpFinder,
-		respWriter:              respWriter,
-		plansConfig:             plansConfig,
-		defaultSubaccountRegion: region,
+		instanceFinder:            instanceFinder,
+		lastOperationFinder:       lastOpFinder,
+		respWriter:                respWriter,
+		plansConfig:               plansConfig,
+		defaultSubaccountRegion:   region,
+		regionParameterIsRequired: regionParameterIsRequired,
```
Comment (piotrmiskiewicz) [inline internal/appinfo/runtime_info.go:48]: you don't need this bool flag

File: internal/appinfo/runtime_info.go [hunk 2]
```diff
@@ -157,7 +159,7 @@ func (h *RuntimeInfoHandler) planNameOrDefault(inst internal.InstanceWithOperati
 	if inst.ServicePlanName != "" {
 		return inst.ServicePlanName
 	}
-	return broker.Plans(h.plansConfig, "", false, false)[inst.ServicePlanID].Name
+	return broker.Plans(h.plansConfig, "", false, false, h.regionParameterIsRequired)[inst.ServicePlanID].Name //tu
```
Comment (piotrmiskiewicz) [inline internal/appinfo/runtime_info.go:162]: you are taking only a name here, the flag which indicates if parameter is required is not necessary here

File: internal/appinfo/runtime_info.go (hunk 1 above)
Comment (MarekMichali) [reply internal/appinfo/runtime_info.go:48]: I will remove it.

File: internal/appinfo/runtime_info.go (hunk 2 above)
Comment (MarekMichali) [reply internal/appinfo/runtime_info.go:162]: You're right, I didn't see that we only take the name parameter.

File: internal/appinfo/runtime_info.go [hunk 3]
```diff
@@ -37,7 +37,7 @@ type RuntimeInfoHandler struct {
 	defaultSubaccountRegion string
 }
 
-func NewRuntimeInfoHandler(instanceFinder InstanceFinder, lastOpFinder LastOperationFinder, plansConfig broker.PlansConfig, region string, respWriter ResponseWriter) *RuntimeInfoHandler {
+func NewRuntimeInfoHandler(instanceFinder InstanceFinder, lastOpFinder LastOperationFinder, plansConfig broker.PlansConfig, region string, respWriter ResponseWriter, regionParameterIsRequired bool) *RuntimeInfoHandler {
```
Comment (piotrmiskiewicz) [inline internal/appinfo/runtime_info.go:40]: regionParameterIsRequired is not used

Comment (MarekMichali) [reply internal/appinfo/runtime_info.go]: Removed.

**Review by piotrmiskiewicz — APPROVED** [review]

**Pull request merged**